packages/table/src/Row.tsx

```diff
@@ -0,0 +1,312 @@
+    if (!columnInfo) {
+      return null;
+    }
+
+    const chevronButton = (
+      <IconButton
+        onClick={() => setIsExpanded(curr => !curr)}
+        aria-label="chevron"
+        className={iconButtonMargin}
+      >
+        <Icon
+          aria-label="chevron"
+          glyph={isExpanded ? 'ChevronDown' : 'ChevronRight'}
+          color={uiColors.gray.dark2}
+        />
+      </IconButton>
+    );
+
+    const nestedRows = React.Children.map(children, child => {
+      if (isComponentType(child, 'Row')) {
+        return React.cloneElement(child, {
+          ref: nodeRef,
+          ['aria-expanded']: isExpanded ? 'true' : 'false',
+          isParentExpanded: isExpanded,
+          indentLevel: indentLevel + 1,
+        });
+      }
+    });
+
+    const renderedChildren = React.Children.map(children, child => {
+      if (isComponentType(child, 'CheckboxCell')) {
+        if (disabled) {
+          return React.cloneElement(child, {
+            className: disabledCell,
+            disabled,
+          });
+        } else {
+          return child;
+        }
+      } else if (isComponentType(child, 'Cell')) {
+        const { className, children } = child.props;
+
+        if (disabled) {
+          return React.cloneElement(child, {
+            className: disabledCell,
+            disabled,
+          });
+        } else {
+          if (nestedRows && nestedRows.length > 0 && !hasSeenFirstCell) {
+            hasSeenFirstCell = true;
+
+            return React.cloneElement(child, {
+              children: (
+                <>
+                  {chevronButton}
+                  <span className={truncation}>{children}</span>
+                </>
+              ),
+              className: cx(displayFlex, className),
+              key: children,
+            });
+          }
+
+          if (!children) {
+            return;
+          }
+
+          return React.cloneElement(child, {
+            children: <span className={truncation}>{children}</span>,
+          });
+        }
+      }
+    });
+
+    const shouldAltRowColor = data && data.length >= 10 && !hasNestedRows;
+
+    const alignmentStyles = Object.entries(
+      columnInfo,
+    ).map(([key, { dataType }]) => styleColumn(key, dataType!));
+
+    return (
+      <>
+        <tr
+          className={cx(
+            rowStyle,
+            getIndentLevelStyle(indentLevel),
+            [...alignmentStyles],
+            {
+              [altColor]: shouldAltRowColor,
+              [disabledStyle]: disabled,
+            },
+            className,
+          )}
+          aria-disabled={disabled}
+          ref={ref}
+          {...rest}
+        >
+          {selectable && (
+            <CheckboxCell
+              index={indexRef.current}
+              className={cx({ [disabledCell]: disabled })}
+              disabled={disabled}
+            />
+          )}
+          {renderedChildren}
+        </tr>
+
+        <Transition
```

Comment:
Can we conditionally render this if nestedRows exist?

Reply:
👍